examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjectsStorage.agent.ts

```diff
@@ -55,7 +86,20 @@ export class BpiSubjectStorageAgent extends PrismaService {
   async updateBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
     const updatedBpiSubjectModel = await this.bpiSubject.update({
       where: { id: bpiSubject.id },
-      data: bpiSubject,
+      data: {
+        id: bpiSubject.id,
+        name: bpiSubject.name,
+        description: bpiSubject.description,
+        loginNonce: bpiSubject.loginNonce,
+        publicKey: bpiSubject.publicKey,
+        roles: {
+          connect: bpiSubject.roles.map((r) => {
+            return {
+              id: r.id,
+            };
+          }),
+        },
+      },
```

Comment:
since we are doing same code in createNewBpiSubject and here, can we either use the mapper here, or extract this out in a function?